exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/Foreman.java

```diff
@@ -853,7 +875,9 @@ public void close() throws Exception {
       // storage write; query completion occurs in parallel with profile
       // persistence.
 
-      queryManager.writeFinalProfile(uex);
+      if (profileOption == ProfileOption.ASYNC) {
+        writeProfile(uex);
```

Comment:
why not call queryManager.writeFinalProfile directly from here and avoid creating another function ?